.github/workflows/build_preview.yml

```diff
@@ -1,13 +1,15 @@
-name: .NET
+name: Build and upload preview build
 
 on:
-  push:
-    branches: [ main ]
   pull_request:
     branches: [ main ]
+    types:
```

Comment:
This seems a little unnecessary. but I like that it makes it clear:  
```
By default, a workflow only runs when a pull_request_target's activity type is opened, 
synchronize, or reopened. To trigger workflows for more activity types, use the types keyword. 
```
https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows